### PR TITLE
feat(react-instantsearch): introduce `<RelatedProducts>` widget

### PR DESCRIPTION
**Pull request by sarahdayan**
This introduces the `<RelatedProducts>` widget.

https://algolia.atlassian.net/browse/FX-2758

### CODE REVIEW

File: packages/react-instantsearch/src/widgets/RelatedProducts.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    { $$widgetType: 'ais.relatedProducts' }
+  );
+
+  const uiProps: UiProps<TItem> = {
+    items: recommendations as Array<Hit<TItem>>,
+    itemComponent,
+    headerComponent,
+    fallbackComponent,
+    status,
+    sendEvent: () => {},
```
Comment (sarahdayan) [inline packages/react-instantsearch/src/widgets/RelatedProducts.tsx:71]: Once the FBT widget is merged we can make `sendEvent` optional in the UI component instead.

**Review by aymeric-giraudet — APPROVED** [review]

File: packages/react-instantsearch/src/widgets/__tests__/RelatedProducts.test.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { createSearchClient } from '@instantsearch/mocks';
+import { InstantSearchTestWrapper } from '@instantsearch/testutils';
+import { render, waitFor } from '@testing-library/react';
+import React from 'react';
+
+import { RelatedProducts } from '../RelatedProducts';
+
+describe('RelatedProducts', () => {
+  test('renders with translations', async () => {
+    const client = createSearchClient();
```
Comment (dhayab) [inline packages/react-instantsearch/src/widgets/__tests__/RelatedProducts.test.tsx:14]: This should probably be using a client with mock recommendations if we want to test translations for the title.

File: packages/react-instantsearch-core/src/lib/useRelatedProducts.ts
Comment (dhayab) [inline packages/react-instantsearch-core/src/lib/useRelatedProducts.ts:1]: Maybe this file should not be here? It looks like an artifact of the parent branch.

**Review by dhayab — CHANGES_REQUESTED** [review]
React widget looks great! I just noticed one issue with a test.

File: packages/react-instantsearch/src/widgets/__tests__/RelatedProducts.test.tsx (hunk 2 above)
Comment (sarahdayan) [reply packages/react-instantsearch/src/widgets/__tests__/RelatedProducts.test.tsx]: Oops, my bad, I overlooked this test.

**Review by dhayab — APPROVED** [review]

**Pull request merged**